applications/StructuralMechanicsApplication/custom_constitutive/truss_plasticity_constitutive_law.cpp

```diff
@@ -0,0 +1,294 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:    Klaus B. Sautter
+//
+// System includes
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "includes/properties.h"
+#include "custom_constitutive/truss_plasticity_constitutive_law.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+//******************************CONSTRUCTOR*******************************************
+//************************************************************************************
+
+TrussPlasticityConstitutiveLaw::TrussPlasticityConstitutiveLaw()
+    : ConstitutiveLaw()
```

Comment:
Can be derived from the elastic one to reduce code duplication?

Reply:
I talked to @philbucher  about inheritance in the constitutive law and I decided to keep them seperated. It is only a few functions so I think it will not make a big difference